### PR TITLE
Add EmacsBehavior to CodeInput and add example

### PR DESCRIPTION
**Pull request by el-ethan**
Have CodeInput inherit EmacsBehavior and add "Key bindings" option to CodeInputTest so that users can try out the bindings.

### CODE REVIEW

File: kivy/uix/codeinput.py [hunk 1]
```diff
@@ -85,6 +85,13 @@ class CodeInput(CodeNavigationBehavior, TextInput):
 
     '''
 
+    active_key_bindings = StringProperty('default')
+    '''String name which determines the type of key bindings to use.
+
+    :attr:`active_key_bindings` is a :class:`~kivy.properties.StringProperty`
+    and defaults to ``'default'``
+    '''
```
Comment (akshayaurora) [inline kivy/uix/codeinput.py]: ..versionadded:: 1.9.2

File: kivy/uix/behaviors/emacs.py [hunk 2]
```diff
@@ -44,6 +47,17 @@ class EmacsBehavior(object):
     keyboard shortcuts for the :class:`~kivy.uix.textinput.TextInput` widget.
     '''
 
+    active_key_bindings = StringProperty('emacs')
+    '''String name which determines the type of key bindings to use with the
+    :class:`~kivy.uix.textinput.TextInput`. This allows Emacs key bindings to
+    be enabled/disabled programmatically for widgets that inherit from
+    :class:`EmacsBehavior`. If the value is not ``'emacs'`` Emacs bindings will
+    be disabled.
+
```
Comment (akshayaurora) [inline kivy/uix/behaviors/emacs.py]: ..versionadded...

**Comment by akshayaurora** [comment]
@el-ethan I think the vim side if this debate is absent... :)

Which behavior to have by default for code input... I feel this would be better served as a garden widget, where we can have EmacsCodeInput VIMCodeinput... instead of limiting to one type of behavior let's keep the core widget open for re-usability.

**Comment by akshayaurora** [comment]
Oh, btw nice job!

**Comment by dessant** [comment]
This repo is a great place for the behavior itself, but i agree with @akshayaurora that we should not enable it for codeinbut by default. Documenting the usage in the behavior docstring is enough.

**Comment by el-ethan** [comment]
@akshayaurora, do you think the vim community would mind if VIMBehavior inherited from EmacsBehavior? Seriously though, I think it would be great if there was a VIMBehavior mixin as well - but better that it be written by a vim user :)

Also, even though CodeInput inherits from EmacsBehavior in this branch, by default the behavior doesn't change from that of kivy master. It will only change if you pass a different value for `active_key_bindings`. 

The reason why I initially added it to the CodeInput is that I was asked to add an example of EmacsBehavior to `examples/widgets/codeinput`, and this was what I came up with at the time. The `active_key_bindings` property allows the behavior of the CodeInput to be changed programmatically so that you don't have to decide what kind of behavior you want to use ahead of time. VIMBehavior could be added to CodeInput later, and used the same way. 

However, I am not so attached to this example, so if you want me to make some changes, or simply close this pull request, it is fine with me. Probably an example is not necessary anyway, since anyone who will use EmacsBehavior is likely already familiar with Emacs.

File: kivy/uix/codeinput.py [hunk 3]
```diff
@@ -52,7 +52,7 @@
 # TODO: color chooser for keywords/strings/...
 
 
-class CodeInput(CodeNavigationBehavior, TextInput):
+class CodeInput(CodeNavigationBehavior, EmacsBehavior, TextInput):
```
Comment (dessant) [inline kivy/uix/codeinput.py]: @el-ethan, we would like to merge your work. Could you remove `EmacsBehavior, ` from this line?

**Comment by el-ethan** [comment]
I removed `EmacsBehavior` from `CodeInput` but I added made a few minor changes so that the example in `examples/widgets/codeinput` still works and can be further extended in the future with VIMBehavior, PicoBehavior ;),  etc. I can still remove the example altogether, if necessary.

**Comment by dessant** [comment]
@el-ethan, thanks a lot! :+1:

**Comment by dessant** [comment]
I've cleaned up a bit in https://github.com/kivy/kivy/pull/3894.

The pr limits EmacsBehavior (feature released in 1.9.1) usage to CodeInput only, before it could also be used with TextInput.

@team, should we move CodeInput.key_bindings to TextInput and preserve the original behavior?

File: examples/widgets/codeinput.py [hunk 4]
```diff
@@ -3,12 +3,13 @@
...
 from kivy.uix.boxlayout import BoxLayout
 from kivy.uix.codeinput import CodeInput
+from kivy.uix.behaviors import EmacsBehavior
 from kivy.uix.popup import Popup
 from kivy.properties import ListProperty
 from kivy.core.window import Window
 from kivy.core.text import LabelBase
 from pygments import lexers
-
+from pygame import font as fonts
```
Comment (tito) [inline examples/widgets/codeinput.py]: This should be removed

**Comment by dessant** [comment]
@tito, it was removed.

**Comment by dessant** [comment]
ref https://github.com/kivy/kivy/pull/3898

**Pull request merged**